crates/compiler/can/src/constraint.rs

```diff
@@ -765,10 +805,20 @@ pub enum Constraint {
     /// The `Slice<Variable>` is used for imports where we manually put the Content into Subs
     /// by copying from another module, but have to make sure that any variables we use to store
     /// these contents are added to `Pool` at the correct rank
-    Let(Index<LetConstraint>, Slice<Variable>),
+    Let(
+        Index<LetConstraint>,
+        Slice<Variable>,
+        Option<Box<Refinements>>,
+    ),
+    LetAndExpandType(
+        Index<LetConstraint>,
+        Slice<Variable>,
+        Option<Box<Refinements>>,
```

Comment:
Is there a reason Refinements is boxed?

Reply:
If I don't box it, one of the size assertions fails for `Constraint`:

```
error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
   --> crates/compiler/can/src/constraint.rs:731:1
    |
731 | roc_error_macros::assert_sizeof_aarch64!(Constraint, 3 * 8);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: source type: `Constraint` (512 bits)
    = note: target type: `[u8; 24]` (192 bits)
    = note: this error originates in the macro `static_assertions::assert_eq_size` which comes from the expansion of the macro `roc_error_macros::assert_sizeof_aarch64` (in Nightly builds, run with -Z macro-backtrace for more info)
```